### PR TITLE
EAG4-20 = Added logic to fetch all players count

### PR DESCRIPTION
**Pull request by Gredgit**

### CODE REVIEW

File: src/screens/home/home.jsx [hunk 1]
```diff
@@ -14,8 +14,12 @@ import './home.scss';
 
 function Homepage() {
   const { gameData, setGameData, playerId } = useContext(GameDataContext);
+  const [playerNum, setPlayerNum] = useState('?');
   const navigate = useNavigate();
-  const playerNum = 97;
+
+  useInterval(async () => {
+    setPlayerNum(await (await getAllPlayersCount(playerId)).data);
```
Comment (MykhailoHoroshko) [inline src/screens/home/home.jsx]: first await is redundant

File: src/screens/home/home.jsx [hunk 2]
```diff
@@ -44,3 +48,22 @@ function Homepage() {
 }
 
 export default Homepage;
+export const useInterval = (callback, delay) => {
```
Comment (MykhailoHoroshko) [inline src/screens/home/home.jsx]: create `/src/hooks/useInterval.js` file and move there

File: src/screens/home/home.jsx [hunk 3]
```diff
@@ -44,3 +48,22 @@ function Homepage() {
...
+  const savedCallback = useRef();
+
+  useEffect(() => {
+    savedCallback.current = callback;
+  }, [callback]);
+
+  useEffect(() => {
+    function tick() {
+      savedCallback.current();
```
Comment (MykhailoHoroshko) [inline src/screens/home/home.jsx]: check if `savedCallback.current !== null` or initialize ref with empty function

**Review by MykhailoHoroshko — CHANGES_REQUESTED** [review]

File: src/screens/home/home.jsx [hunk 4]
```diff
@@ -44,3 +48,22 @@ function Homepage() {
 }
 
 export default Homepage;
+export const useInterval = (callback, delay) => {
+  const savedCallback = useRef();
+
+  useEffect(() => {
+    savedCallback.current = callback;
```
Comment (MykhailoHoroshko) [inline src/screens/home/home.jsx]: add check if callback is function

Comment (Gredgit) [reply src/screens/home/home.jsx]: fixed

File: src/hooks/useInterval.js [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import { useRef, useEffect } from 'react';
+export const useInterval = (callback, delay) => {
```
Comment (MykhailoHoroshko) [inline src/hooks/useInterval.js]: Change to default export

**Review by MykhailoHoroshko — APPROVED** [review]

**Pull request merged**